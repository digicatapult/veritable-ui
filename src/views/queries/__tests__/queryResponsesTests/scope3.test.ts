import { expect } from 'chai'
import { describe } from 'mocha'
import Scope3CarbonConsumptionResponseTemplates from '../../queryResponses/respondToScope3.js'
import Scope3CarbonConsumptionViewResponseTemplates from '../../queryResponses/viewResponseToScope3.js'

describe('QueryResponseTemplates', () => {
  describe('newScope3CarbonConsumptionResponseFormPage Tests', () => {
    it('should render with test data', async () => {
      const templates = new Scope3CarbonConsumptionResponseTemplates()
      const rendered = await templates.newScope3CarbonConsumptionResponseFormPage(
        'form',
        {
          company_name: 'VER123',
          company_number: '3456789',
          status: 'verified_both',
          id: '11',
          agent_connection_id: 'agentId',
          pin_tries_remaining_count: null,
          pin_attempt_count: 0,
          created_at: new Date(),
          updated_at: new Date(),
        },
        'xyz1234',
        2,
        'cg34jdt'
      )
      expect(rendered).to.matchSnapshot()
    })
    it('should escape html in name', async () => {
      const templates = new Scope3CarbonConsumptionResponseTemplates()
      const rendered = await templates.newScope3CarbonConsumptionResponseFormPage(
        'form',
        {
          company_name: '<div>I own you</div>',
          company_number: '3456789',
          status: 'verified_both',
          id: '11',
          agent_connection_id: 'agentId',
          pin_tries_remaining_count: null,
          pin_attempt_count: 0,
          created_at: new Date(),
          updated_at: new Date(),
        },
        'xyz1234',
        2,
        'cg34jdt'
      )
      expect(rendered).to.matchSnapshot()
    })
  })
  describe('newQuerySuccess Tests', () => {
    it('should render sucess query response page', async () => {
      const templates = new Scope3CarbonConsumptionResponseTemplates()
      const rendered = await templates.newScope3CarbonConsumptionResponseFormPage('success', {
        company_name: 'VER123',
        company_number: '3456789',
        status: 'verified_both',
        id: '11',
        agent_connection_id: 'agentId',
        pin_tries_remaining_count: null,
        pin_attempt_count: 0,
        created_at: new Date(),
        updated_at: new Date(),
      })
      expect(rendered).to.matchSnapshot()
    })

    it('should escape html in name', async () => {
      const templates = new Scope3CarbonConsumptionResponseTemplates()
      const rendered = await templates.newScope3CarbonConsumptionResponseFormPage('success', {
        company_name: '<div>I own you</div>',
        company_number: '3456789',
        status: 'verified_both',
        id: '11',
        agent_connection_id: 'agentId',
        pin_tries_remaining_count: null,
        pin_attempt_count: 0,
        created_at: new Date(),
        updated_at: new Date(),
      })
      expect(rendered).to.matchSnapshot()
    })
  })
  describe('viewing query response Tests', () => {
    const sampleDate = new Date(Date.UTC(2024, 6, 4))
    it('should render resolved query to view response', async () => {
      const templates = new Scope3CarbonConsumptionViewResponseTemplates()
      const rendered = await templates.scope3CarbonConsumptionViewResponsePage({
        id: '11',
        company_name: 'VER123',
        query_type: 'Scope 3 query',
        updated_at: sampleDate,
        status: 'resolved',
        role: 'requester',
        quantity: '430',
        productId: 'jkl333',
        emissions: '5678',
      })
      expect(rendered).to.matchSnapshot()
    })
    it('shouldexcape html resolved query to view response', async () => {
      const templates = new Scope3CarbonConsumptionViewResponseTemplates()
      const rendered = await templates.scope3CarbonConsumptionViewResponsePage({
        id: '11',
        company_name: '<div>VER123</div>',
        query_type: 'Scope 3 query',
        updated_at: sampleDate,
        status: 'resolved',
        role: 'requester',
        quantity: '<div>430</div>',
        productId: 'jkl333',
        emissions: '5678',
      })
      expect(rendered).to.matchSnapshot()
    })
  })
})
