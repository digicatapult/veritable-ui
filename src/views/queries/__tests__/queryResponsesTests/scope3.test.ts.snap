// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryResponseTemplates newQuerySuccess Tests should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Select Company</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Select Company\\" href=\\"/queries/scope-3-carbon-consumption-response\\">Select Company</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div id=\\"new-query-confirmation-text\\"><p>Your query Response has been shared with the following company: test name.</p><p>Thank you for answering this query, there is no other action required.</p><a class=\\"button \\" href=\\"/\\" data-variant=\\"filled\\"><span>Back to Home</span></a></div></div></div></main></body></html>"`;

exports[`QueryResponseTemplates newQuerySuccess Tests should render sucess query response page 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Select Company</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Select Company\\" href=\\"/queries/scope-3-carbon-consumption-response\\">Select Company</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div id=\\"new-query-confirmation-text\\"><p>Your query Response has been shared with the following company: VER123.</p><p>Thank you for answering this query, there is no other action required.</p><a class=\\"button \\" href=\\"/\\" data-variant=\\"filled\\"><span>Back to Home</span></a></div></div></div></main></body></html>"`;

exports[`QueryResponseTemplates newScope3CarbonConsumptionResponseFormPage Tests should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Select Company</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Select Company\\" href=\\"/queries/scope-3-carbon-consumption-response\\">Select Company</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"container-scope3-carbon\\"><div class=\\"box1\\"><h1>Scope 3 Carbon Consumption</h1><p class=\\"query-text-carbon3-consumption\\">Provide the total scope 3 carbon consumption for the specified products / component. If you do not have all the required information, please forward this query to your suppliers, to aggregate their responses before submitting the final total. </p></div><div class=\\"box2\\"><p>What are the total Scope 3 carbon emissions for the product/component below? </p><form id=\\"scope-3-carbon-consumption\\" hx-post=\\"/queries/scope-3-carbon-consumption-response/submit\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\"><p>Product ID: cg34jdt</p><p>Quantity: 2</p><input type=\\"hidden\\" name=\\"companyId\\" value=\\"11\\"/><input type=\\"hidden\\" name=\\"queryId\\" value=\\"xyz1234\\"/><div class=\\"input-container\\"><label for=\\"total-scope-3-carbon-emissions-input\\" class=\\"input-label\\">Total Scope 3 carbon emissions</label><input id=\\"total-scope-3-carbon-emissions-input\\" name=\\"totalScope3CarbonEmissions\\" placeholder=\\"Value in kg CO2e (to be aggregated)\\" class=\\"query-input-field\\" type=\\"text\\" required/></div><div class=\\"input-container\\"><label for=\\"partial-response-input\\">Partial response</label><input id=\\"partial-response-input\\" name=\\"partialResponse\\" type=\\"checkbox\\"/></div><p>*If partial response checkbox is ticked, you must share this query with another supplier in your network, where your responses will be aggregated.</p><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"success\\"><span>Submit Query</span></button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryResponseTemplates newScope3CarbonConsumptionResponseFormPage Tests should render with test data 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Select Company</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Select Company\\" href=\\"/queries/scope-3-carbon-consumption-response\\">Select Company</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"container-scope3-carbon\\"><div class=\\"box1\\"><h1>Scope 3 Carbon Consumption</h1><p class=\\"query-text-carbon3-consumption\\">Provide the total scope 3 carbon consumption for the specified products / component. If you do not have all the required information, please forward this query to your suppliers, to aggregate their responses before submitting the final total. </p></div><div class=\\"box2\\"><p>What are the total Scope 3 carbon emissions for the product/component below? </p><form id=\\"scope-3-carbon-consumption\\" hx-post=\\"/queries/scope-3-carbon-consumption-response/submit\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\"><p>Product ID: cg34jdt</p><p>Quantity: 2</p><input type=\\"hidden\\" name=\\"companyId\\" value=\\"11\\"/><input type=\\"hidden\\" name=\\"queryId\\" value=\\"xyz1234\\"/><div class=\\"input-container\\"><label for=\\"total-scope-3-carbon-emissions-input\\" class=\\"input-label\\">Total Scope 3 carbon emissions</label><input id=\\"total-scope-3-carbon-emissions-input\\" name=\\"totalScope3CarbonEmissions\\" placeholder=\\"Value in kg CO2e (to be aggregated)\\" class=\\"query-input-field\\" type=\\"text\\" required/></div><div class=\\"input-container\\"><label for=\\"partial-response-input\\">Partial response</label><input id=\\"partial-response-input\\" name=\\"partialResponse\\" type=\\"checkbox\\"/></div><p>*If partial response checkbox is ticked, you must share this query with another supplier in your network, where your responses will be aggregated.</p><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"success\\"><span>Submit Query</span></button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryResponseTemplates should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Select Company</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Select Company\\" href=\\"/queries/scope-3-carbon-consumption-response\\">Select Company</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div id=\\"new-query-confirmation-text\\"><p>Your query Response has been shared with the following company: &lt;div>I own you&lt;/div>.</p><p>Thank you for answering this query, there is no other action required.</p><a class=\\"button \\" href=\\"/\\" data-variant=\\"filled\\"><span>Back to Home</span></a></div></div></div></main></body></html>"`;
